src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +805,113 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend if
+ * it is not already assigned
+ */
+void
+AssignGlobalPID(void)
+{
+	if (MyBackendData->globalPID != 0)
+	{
+		return;
+	}
+
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		uint64 globalPID = GenerateGlobalPID();
+
+		SpinLockAcquire(&MyBackendData->mutex);
+		MyBackendData->globalPID = globalPID;
+		SpinLockRelease(&MyBackendData->mutex);
+
+		return;
+	}
+
+	uint64 globalPID = ExtractGlobalPID();
```

Comment:
Minor:
For example, `ExtractGlobalPID(application_name)` would be much nicer to follow the code.

Also,  adding an `else` seems to increase readability `if( IsCitusInitiatedRemoteBackend) (Extrat) else {Generate}`